lib/apartment/adapters/sqlite3_adapter.rb

```diff
@@ -13,6 +13,9 @@ def initialize(config)
         @default_dir = File.expand_path(File.dirname(config[:database]))
 
         super
+
+        Apartment.connection_handler = ActiveRecord::ConnectionAdapters::ConnectionHandler.new
+        reset
```

Comment:
It's only relevant to connection based adapters. Schema based adapters don't need a new connection handler. So from an abstract sense, it doesn't apply.

The purpose btw of the abstract adapter is not just removing duplication. It's actually defining the public interface that all adapters are expected to follow. Subclasses can override certain behaviours to their own desire, but something such as modifying connection handlers is probably only relevant to connection based adapters.